### PR TITLE
feat: implement email service with Brevo integration

### PR DESCRIPTION
**Pull request by Evil-star25**
## WHY
 - https://github.com/PkFokam-Alumni-Network/CommunityBackendService/issues/71

## WHAT

- Created `email_service.py` to send announcement emails via Brevo API
- Added `test_email_service.py` to test email sending logic, including edge cases
- Updated `announcement_router.py` to trigger emails on announcement creation
- Extended `settings.py` to include Brevo-related configuration (sender name, email, API key)
- Added `settings_test.py` for configuration testing
- Updated `requirements.txt` to include `sib-api-v3-sdk` and related dependencies

### CODE REVIEW

File: services/email_service.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        self.sender_name = settings.ADMIN_NAME
+        self.sender_email = settings.ADMIN_EMAIL
+        self.api_key = settings.BREVO_API_KEY  
+
+        configuration = sib_api_v3_sdk.Configuration()
+        configuration.api_key['api-key'] = self.api_key
+        self.api_client = sib_api_v3_sdk.ApiClient(configuration)
+        self.email_api = sib_api_v3_sdk.TransactionalEmailsApi(self.api_client)
+
+    def get_all_user_emails(self, db: Session) -> List[str]:
```
[the remark this answers is not in the document]
Comment (bjoelfra) [reply services/email_service.py:22]: All of this new service is not necessary you can just crete an endpoint to do this, you can follow steps similar to the one we used implementing sendGrid in the utils functions

**Comment by bjoelfra** [comment]
We also need to create an endpoint to send an email to only one user. An alternative to always querying the db for user emails is to keep a dynamic list which is populated each time a new user is created and to avoid redundancy you can make a check if the email you are adding is already present in the db. That will be more efficient in my opinion

File: services/email_service.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bjoelfra) [reply services/email_service.py:22]: and you can rename it to get_users_emails

[the remark this answers is not in the document]
Comment (bjoelfra) [reply services/email_service.py:22]: To follow clean architecture best practices, all database access should be handled in the repository layer, not the service layer.

**Comment by bjoelfra** [comment]
Please make sure to expose actual REST endpoints for this functionality so it can be accessed via the API. Once that's done, test the endpoints using Postman to ensure they're working as expected and returning the correct responses.

File: services/email_service.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Evil-star25) [reply services/email_service.py:22]: got it

File: services/email_service.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        users = db.query(User).all()
+        return [user.email for user in users if user.email]
+
+    def send_announcement_email(self, db: Session, announcement: Announcement) -> None:
```
[the remark this answers is not in the document]
Comment (Evil-star25) [reply services/email_service.py:26]: okay. I have refactored it using jinja2 and created a template for the announcement email

File: routers/announcement_router.py [hunk 3]
```diff
@@ -5,15 +5,19 @@
...
 router = APIRouter()
 
 @router.post("/announcements/", status_code=status.HTTP_201_CREATED, response_model=AnnouncementResponse)
 def create_announcement(announcement: AnnouncementCreate, session: Session = Depends(get_db)) -> AnnouncementResponse:
     service = AnnouncementService()
+    email_service = EmailService()
     try:
         ann = service.create_announcement(session, announcement)
+
+        email_service.send_announcement_email(session, ann)
```
[the remark this answers is not in the document]
Comment (Evil-star25) [reply routers/announcement_router.py:20]: What if we make use of cohere api to summarize the announcement dynamically?

File: routers/user_router.py [hunk 4]
```diff
@@ -191,4 +193,19 @@ def reset_password(body: user_schema.PasswordReset,
         except Exception as e:
             LOGGER.error(f"SERVER ERROR in reset_password: {str(e)}")
             raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail="Internal server error")
-        
\ No newline at end of file
+
+router.post("/users/{user_id}/send-announcement",  status_code=status.HTTP_200_OK, response_model=user_schema.UserUpdate)
```
Comment (bjoelfra) [inline routers/user_router.py:197]: You might want to run this as a backgroundTask.

File: services/email_service.py [hunk 5]
```diff
@@ -19,35 +22,47 @@ def __init__(self):
         self.api_client = sib_api_v3_sdk.ApiClient(configuration)
         self.email_api = sib_api_v3_sdk.TransactionalEmailsApi(self.api_client)
 
-    def get_all_user_emails(self, db: Session) -> List[str]:
-        users = db.query(User).all()
-        return [user.email for user in users if user.email]
+        self.user_repo = UserRepository()
+        
+        self.template_env = Environment(
+            loader=FileSystemLoader(searchpath=os.path.join(os.getcwd(), "templates")),
```
Comment (bjoelfra) [inline services/email_service.py:28]: Avoid using `os.getcwd()`,  it assumes your project will always be ran from your root directory.
Use `__file__` and `pathlib`, as it is independent of the project directory

File: services/email_service.py [hunk 6]
```diff
@@ -56,3 +71,31 @@ def send_announcement_email(self, db: Session, announcement: Announcement) -> No
         except Exception as e:
             LOGGER.error(f"Unexpected error in sending email: {e}")
             raise
+
+    def send_email_to_user(self, user: User, announcement: Announcement) -> None:
```
Comment (bjoelfra) [inline services/email_service.py:75]: What is the difference betweent this and send_announcement_email ?

Comment (Evil-star25) [reply services/email_service.py:75]: This was because you said we wanted to test for just one user. not all

**Comment by Evil-star25** [comment]
I should delete setting.py?

**Comment by Evil-star25** [comment]
I am lost when pulling I see many conflicts
